src/hooks/useAccess.ts

```diff
@@ -1,25 +1,56 @@
 import { RequestBundle, apiRequest } from './apiRequest';
+import { FlashContext } from '../contexts/FlashContext';
+import { useContext } from 'react';
 
-const ACCESS_BASE_URL = process.env.NEXT_PUBLIC_ACCESS_BASE_URL;
 
 /**
  * Custom hook for making API requests to the access API
- * @returns Object containing request method for making API calls
+ * @returns Object containing a request method for making API calls
  */
 export const useAccess = () => {
+  const flashContext = useContext(FlashContext);
+
+  if (!flashContext) {
+    throw new Error('useAccess must be used within a FlashProvider');
+  }
+  const { processFlash } = flashContext;
+
   /**
-   * @typeParam T - The type of the response data
-   * @param requestBundle - Bundle containing endpoint and request options
-   * @throws An error When the API request fails
-   * @returns A promise resolving to the response data
-   * @returns A Promise resolving to the response data or an error
+   * Processes API errors and extracts flash messages
+   * @param error - The API error
+   * @returns The extracted flash message or a default error message
+   */  
+  const handleError = (error: any) => {
+    try {
+      const errorResponse = JSON.parse(error.message);
+      return errorResponse?.flash ? errorResponse : { flash: { error: ['An unexpected error occurred.'] } };
+    } catch {
+      return { flash: { error: ['An unexpected error occurred.'] } };
+    }
+  };
+
+  /**
+   * Makes an API request and processes flash messages from the response.
+   * @typeParam T - The expected response data type.
+   * @param requestBundle - Bundle containing endpoint and request options.
+   * @returns A Promise resolving to the response data.
+   * @throws An error if the API request fails.
    */
-  const request = async <T>(requestBundle: RequestBundle): Promise<T> => {
-    const accessUri = `${ACCESS_BASE_URL}${requestBundle.endpoint}`;
+  const request = async <T extends object>(requestBundle: RequestBundle): Promise<T> => {
+    const accessUri = `${process.env.NEXT_PUBLIC_ACCESS_BASE_URL}${requestBundle.endpoint}`;
```

Comment:
You need to check if ACCESS_BASE_URL exists before using it. If an error occurs because it doesn't exist and it's not thrown from here, it's going to be hard to figure out where the error is coming from.

```suggestion
    const accessUri = `${ACCESS_BASE_URL}${requestBundle.endpoint}`;
```